File: android/sample-ui/src/main/kotlin/com/usesmileid/sampleapps/ui/components/UseSmileIDSampleStatusBadge.kt
package com.usesmileid.sampleapps.ui.components

import androidx.compose.foundation.background
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.unit.sp
import com.smileid.designsystem.SmileDimens
import com.smileid.designsystem.smileLabelTracking
import com.smileid.designsystem.smileLabelSize
import com.usesmileid.sampleapps.ui.model.UseSmileIDSampleStatus
import com.usesmileid.sampleapps.ui.theme.UseSmileIDSampleTheme

/** A status pill drawn with the design's soft tinted fills (`softBadgeTokens()`, spec/design-tokens.json → softBadgeFills). */
@Composable
fun UseSmileIDSampleStatusBadge(
    status: UseSmileIDSampleStatus,
    modifier: Modifier = Modifier,
    testId: String? = null,
) {
    val badge = UseSmileIDSampleTheme.colors.badge
    val (background, foreground) = when (status) {
        UseSmileIDSampleStatus.Clear -> badge.successBackground to badge.successText
        UseSmileIDSampleStatus.Attention -> badge.warningBackground to badge.warningText
        UseSmileIDSampleStatus.Blocked -> badge.errorBackground to badge.errorText
        UseSmileIDSampleStatus.Processing -> badge.infoBackground to badge.infoText
    }
    StatusPill(label = status.label, background = background, foreground = foreground, modifier = modifier, testId = testId)
}

@Composable
private fun StatusPill(
    label: String,
    background: Color,
    foreground: Color,
    modifier: Modifier,
    testId: String?,
) {
    Text(
        text = label,
        style = UseSmileIDSampleTheme.type.textStyleOverline.copy(
            fontSize = smileLabelSize,
            letterSpacing = smileLabelTracking,
        ),
        color = foreground,
        modifier = modifier
            .tagged(testId)
            .background(color = background, shape = RoundedCornerShape(SmileDimens.radiusControl))
            .padding(horizontal = SmileDimens.spacingXs, vertical = SmileDimens.space4),
    )
}

File: ios/SampleUI/Sources/SampleUI/Components/UseSmileIDSampleStatusBadge.swift
import SwiftUI

/// A soft tinted status pill: the label is Title case, not upper-cased like the same style's headers, and it takes `radius.control`.
public struct UseSmileIDSampleStatusBadge: View {
  private let status: UseSmileIDSampleStatus
  private let testId: String?

  @Environment(\.useSmileIDSampleColors) private var colors

  public init(status: UseSmileIDSampleStatus, testId: String? = nil) {
    self.status = status
    self.testId = testId
  }

  public var body: some View {
    UseSmileIDSampleText(
      status.label,
      style: UseSmileIDSampleTheme.type.textStyleOverline
        .with(size: smileLabelSize, tracking: smileLabelTracking)
    )
    .foregroundColor(fill.foreground)
    .padding(.horizontal, SmileSpacing.spacingXs)
    .padding(.vertical, SmileSpacing.space4)
    .background(
      RoundedRectangle(cornerRadius: UseSmileIDSampleShapes.pill, style: .continuous)
        .fill(fill.background)
    )
    .useSmileIDSampleTestId(testId)
  }

  private var fill: (background: Color, foreground: Color) {
    let badge = colors.badge
    switch status {
    case .clear: return (badge.successBackground, badge.successText)
    case .attention: return (badge.warningBackground, badge.warningText)
    case .blocked: return (badge.errorBackground, badge.errorText)
    case .processing: return (badge.infoBackground, badge.infoText)
    }
  }
}
